public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboardUtils.ts

```diff
@@ -50,14 +52,20 @@ export const publicDashboardPersisted = (publicDashboard?: PublicDashboard): boo
 /**
  * Get unique datasource names from all panels that are not currently supported by public dashboards.
  */
-export const getUnsupportedDashboardDatasources = (panels: PanelModel[]): string[] => {
+export const getUnsupportedDashboardDatasources = async (panels: PanelModel[]): Promise<string[]> => {
   let unsupportedDS = new Set<string>();
 
   for (const panel of panels) {
     for (const target of panel.targets) {
-      let ds = target?.datasource?.type;
-      if (ds && !supportedDatasources.has(ds)) {
-        unsupportedDS.add(ds);
+      const dsType = target?.datasource?.type;
+      if (dsType && supportedDatasources.has(dsType)) {
+        const ds = await getDatasourceSrv().get(target.datasource);
+        if (!(ds instanceof DataSourceWithBackend)) {
```

Comment:
This new logic will not add "Cloudwatch" as unsupported. Should we have a manual list or another source of truth?